src/reflect/scala/reflect/internal/transform/UnCurry.scala

```diff
@@ -70,7 +70,8 @@ trait UnCurry {
   object DesugaredParameterType {
     def isUnboundedGeneric(tp: Type) = tp match {
       case t @ TypeRef(_, sym, _) if sym.isAbstractType =>
-        sym.info.resultType.bounds.emptyUpperBound
+        val hi = sym.info.resultType.upperBound
+        typeIsAny(hi) || hi.isWildcard
```

Comment:
Why make `emptyUpperBound` private, and then copy its rhs here?

Reply:
Because `emptyUpperBound` is an instance method of the `TypeBounds` class, and since it is creating that instance that I want to avoid. As an alternative, we could re-define the method as a function, close to the `TypeBounds` class. For instance, in its companion object:  

```Scala
def isEmptyUpperBound(hi: Type): Boolean = typeIsAny(hi) || hi.isWildcard
```

Reply:
Ah, of course; I got lost in `Types.scala` and thought it was on every `Type`.